communication/inc/coap_channel_new.h

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2023 Particle Industries, Inc.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <memory>
+#include <cstdint>
+
+#include "message_channel.h"
+#include "coap_api.h"
+#include "coap.h" // For token_t
+
+#include "system_tick_hal.h"
+
+#include "ref_count.h"
+
+namespace particle::protocol {
+
+class CoapMessageDecoder;
+class Protocol;
+
+namespace experimental {
+
+// This class implements the new experimental protocol API that allows the system to interact with
```

Comment:
Can we remove the experimental namespace? I'd rather have an explanation like "this supercedes the CoAP API xyz and should be used for new message types" than one that says "new and experimental". It won't always be new and if we're using it for real messages, it's not experimental anymore.

Reply:
OK. You can make the final decision here on wheter to change and to what.

Reply:
I'll keep `experimental` until we have proper unit tests for this API.